homeassistant/components/mystrom/light.py

```diff
@@ -42,6 +47,23 @@
 )
 
 
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the myStrom entities."""
+    info = hass.data[DOMAIN][entry.entry_id]["info"]
+    device_type = info["type"]
+
+    if device_type == 102:
+        device = hass.data[DOMAIN][entry.entry_id]["device"]
+        try:
+            await device.get_state()
+        except MyStromConnectionError as err:
+            _LOGGER.error("No route to myStrom bulb: %s", info["ip"])
+            raise ConfigEntryNotReady() from err
```

Comment:
You should move this check to `__init__` instead.
You can raise `ConfigEntryNotReady` there, and when you reach the platform you don't need to worry about it anymore.